test/fw/ptl/utils/pbs_snaputils.py

```diff
@@ -0,0 +1,1675 @@
+
+        if self.create_tar:
+            return self.outtar_path
+        else:
+            return self.snapdir
+
+    def capture_pbs_info(self):
+        """
+        Capture additional metadata on PBSPro
+
+        :returns: name of the output directory/tarfile containing the snapshot
+        """
+        self.logger.info("capturing additional PBS information")
+
+        # Go through 'pbs_info' and capture info that depends on commands
+        for (path, cmd_list) in self.pbs_info.values():
+            if cmd_list is None:
+                continue
+            cmd_list_cpy = list(cmd_list)
+
+            # Add the path to PBS_EXEC to the command path
+            # The command path is the first entry in command list
+            cmd_list_cpy[0] = os.path.join(self.pbs_exec, cmd_list[0])
+            if "pbs_hostn" in cmd_list[0]:
+                # Append hostname to the command list
+                cmd_list_cpy.append(self.server_host)
+            snap_path = os.path.join(self.snapdir, path)
+            self.__capture_cmd_output(self.server_host, snap_path,
+                                      cmd_list_cpy)
+
+        # Capture pbs.conf
+        self.logger.debug("capturing pbs.conf")
+        snap_confpath = os.path.join(self.snapdir, PBS_CONF_PATH)
+        with open(snap_confpath, "w") as fd:
+            for k, v in self.server.pbs_conf.items():
+                fd.write(k + "=" + str(v) + "\n")
+
+        # Capture pbs_environment
+        self.logger.debug("capturing pbs_environment")
+        snap_envpath = os.path.join(self.snapdir, PBS_ENV_PATH)
+        if self.server.pbs_env is not None:
+            with open(snap_envpath, "w") as envfd:
+                for k, v in self.server.pbs_env.iteritems():
+                    envfd.write(k + "=" + v + "\n")
+
+        if self.create_tar:
+            self.__add_to_archive(snap_confpath)
+            self.__add_to_archive(snap_envpath)
+            return self.outtar_path
+        else:
+            return self.snapdir
+
+    def capture_system_info(self):
+        """
+        Capture system related information
+
+        :returns: name of the output directory/tarfile containing the snapshot
+        """
+        self.logger.info("capturing system information")
+
+        # Go over all the hosts specified
+        for host in self.all_hosts:
+            host_platform = self.du.get_platform(host)
+
+            # Capture platform dependent information
+            if host_platform.startswith("linux"):
+                # Go through 'sys_info' and capture info that depends on
+                # commands
+                for (path, cmd_list) in self.sys_info.values():
+                    if cmd_list is None:
+                        continue
+                    cmd_list_cpy = list(cmd_list)
+
+                    # Find the full path to the command on the host
+                    cmd_full = self.du.which(host, cmd_list[0])
```

Comment:
That seems bad design to me, it should return None if it can't find it unless I'm missing something here, I've filed a ticket to fix it: https://pbspro.atlassian.net/browse/PP-831